src/components/ProductCard/ProductCard.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+
+import { BasketCard } from '@/components/BasketCard/BasketCard.tsx';
+import styles from '@/components/ProductCard/ProductCard.module.css';
+import ProductCards from '@/productsList.json';
+
+interface BasketItem {
+    id: number;
+    quantity: number;
+    length: number;
+}
+
+export const ProductCard = () => {
+    const [, setBasketItems] = useState<BasketItem[]>([]);
+    const [productQuantities, setProductQuantities] = useState<Record<number, number>>({});
+    const addToBasket = (productId: number) => {
+        const existingBasket = JSON.parse(localStorage.getItem('basket') || '[]');
+        const currentQuantity = productQuantities[productId] || 1;
+
+        const itemExists = existingBasket.find((item: { id: number }) => item.id === productId);
+
+        if (itemExists) {
+            const updatedBasket = existingBasket.map((item: { id: number; quantity: number }) => {
+                if (item.id === productId) {
+                    return { ...item, quantity: item.quantity + 1 };
+                }
+                return item;
+            });
+
+            setBasketItems(updatedBasket);
+            localStorage.setItem('basket', JSON.stringify(updatedBasket));
+            setProductQuantities({ ...productQuantities, [productId]: currentQuantity + 1 });
+        } else {
+            const newBasketItem = { id: productId, quantity: 1 };
+            setBasketItems([...existingBasket, newBasketItem]);
+            localStorage.setItem('basket', JSON.stringify([...existingBasket, newBasketItem]));
+        }
+    };
+
+    return (
+        <main className={styles.productsContainer}>
+            {ProductCards.map((cards) => (
```

Comment:
You need to do this "map" in your "ProductList" component

"ProductCard" component should contain one item (product card)

This is your layout for "ProductCard":
![image](https://github.com/DavidTereshchenko/fe-react-2024/assets/108006836/ae7e4ba7-306c-4c9d-894e-1c57ef3cb88a)